File: Lista de Exercicio Paradigmas/Lista Java/Ex3.java
public class ContaBancaria {
    private String titular;
    private double saldo;

    public ContaBancaria(String titular) {
        this.titular = titular;
        this.saldo = 0;
    }

    public void depositar(double valor) {
        if (valor > 0) {
            this.saldo += valor;
            System.out.println("Depósito de R$ " + String.format("%.2f", valor) + " realizado. Saldo atual: R$ " + String.format("%.2f", this.saldo) + ".");
        } else {
            System.out.println("Valor de depósito inválido.");
        }
    }

    public void sacar(double valor) {
        if (0 < valor && valor <= this.saldo) {
            this.saldo -= valor;
            System.out.println("Saque de R$ " + String.format("%.2f", valor) + " realizado. Saldo atual: R$ " + String.format("%.2f", this.saldo) + ".");
        } else if (valor > this.saldo) {
            System.out.println("Saldo insuficiente para saque.");
        } else {
            System.out.println("Valor de saque inválido.");
        }
    }

    public void exibirSaldo() {
        System.out.println("Saldo atual: R$ " + String.format("%.2f", this.saldo) + ".");
    }

    public static void main(String[] args) {
        ContaBancaria conta1 = new ContaBancaria("João Silva");

        conta1.depositar(100);
        conta1.exibirSaldo();

        conta1.sacar(30);
        conta1.exibirSaldo();

        conta1.sacar(80);
        conta1.exibirSaldo();
    }
}

File: Lista de Exercicio Paradigmas/Lista Python/Ex3.py
class ContaBancaria:
    def __init__(self, titular):
        self.titular = titular
        self.__saldo = 0  

    def depositar(self, valor):
        if valor > 0:
            self.__saldo += valor
            print(f'Depósito de R$ {valor:.2f} realizado. Saldo atual: R$ {self.__saldo:.2f}.')
        else:
            print('Valor de depósito inválido.')

    def sacar(self, valor):
        if 0 < valor <= self.__saldo:
            self.__saldo -= valor
            print(f'Saque de R$ {valor:.2f} realizado. Saldo atual: R$ {self.__saldo:.2f}.')
        elif valor > self.__saldo:
            print('Saldo insuficiente para saque.')
        else:
            print('Valor de saque inválido.')

    def exibir_saldo(self):
        print(f'Saldo atual: R$ {self.__saldo:.2f}.')

conta1 = ContaBancaria('João Silva')

conta1.depositar(100)
conta1.exibir_saldo()

conta1.sacar(30)
conta1.exibir_saldo()

conta1.sacar(80)
conta1.exibir_saldo()